File: examples/java-transformers/simple-library/src/main/java/com/apple/examples/transformers/Echo.java
package com.apple.examples.transformers;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.lang.invoke.MethodHandles;
import java.util.Map;
import java.util.function.Function;

public class Echo implements Function<Map<String, Object>, Map<String, Object>> {
    private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());

    @Override
    public Map<String, Object> apply(final Map<String, Object> input) {
        LOGGER.info("Echo apply method invoked with input - %s", input);
        return input;
    }
}

File: examples/python-transformers/non-embedded/src/foo/bar/echo.py
import logging

from stargate.api.transform_api import Transform

logging.basicConfig(encoding='UTF-8', level=logging.INFO)
logger = logging.getLogger(__name__)


class Echo(Transform):
    def __init__(self):
        logging.info("Initialized Echo class successfully")

    def apply(self, input: dict):
        logger.info("Echo apply method invoked with input - " + input.__str__())
        return input
